### PR TITLE
Introducing AJAX to project.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
The following action will be completed using AJAX to make site more responsive:

* While user is specifying his/her username, the entered username would be checked for duplicates already present.

### CODE REVIEW

**Comment by saadyousafarbi** [comment]
The javascript file ```jquery-3.2.1.slim.min.js``` does not support AJAX function hence I had to download and include the ```jquery-3.1.0.min.js``` file to get the required AJAX functionality.

File: templates/base.html [hunk 1]
```diff
@@ -19,8 +19,10 @@ <h3 class="text-center">A website with information about automobiles for buying/
     {% block content %}{% endblock content %}
     </body>
 
-<script src="{% static 'js/jquery-3.2.1.slim.min.js' %}"></script>
+<!-- <script src="{% static 'jquery-3.2.1.slim.min.js' %}"></script> -->
+<script src="{% static 'js/jquery-3.1.0.min.js' %}"></script>
```
Comment (zubair-arbi) [inline templates/base.html]: why use older version of jquery?

File: templates/register.html [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+          'username': username
+        },
+        dataType: 'json',
+        success: function (data) {
+          if (data.is_taken) {
+            alert("A user with this username already exists.");
+          }
+        }
+      });
+     $("#id_username").val('');
```
Comment (zubair-arbi) [inline templates/register.html]: no need to reset form field value. It's upto the user to change the form inputs value.

**Review by zubair-arbi — APPROVED** [review]
some minor suggestions otherwise 👍

File: templates/register.html (hunk 2 above)
Comment (saadyousafarbi) [reply templates/register.html]: ok

File: templates/base.html (hunk 1 above)
Comment (saadyousafarbi) [reply templates/base.html]: the version we had previously does not support AJAX.

Comment (saadyousafarbi) [reply templates/base.html]: we previously had the slim version of jQuery which does not support AJAX.
https://stackoverflow.com/questions/44212202/my-javascript-is-returning-this-error-ajax-is-not-a-function

**Comment by saadyousafarbi** [comment]
Rebased 840626c4e070fe2e72f357c01e7d062ccdaee174

**Pull request merged**